src/jit/hwintrinsicxarch.cpp

```diff
@@ -265,6 +265,39 @@ bool Compiler::compSupportsHWIntrinsic(InstructionSet isa)
                                                     isFullyImplmentedISAClass(isa));
 }
 
+//------------------------------------------------------------------------
+// impUnsupportedHWIntrinsic: returns a node for an unsupported HWIntrinsic
+//
+// Arguments:
+//    method     -- method handle of the intrinsic function.
+//    sig        -- signature of the intrinsic call
+//    mustExpand -- true if the intrinsic must return a GenTree*; otherwise, false
+//
+// Return Value:
+//    a gtNewMustThrowException if mustExpand is true; otherwise, nullptr
+//
+GenTree* Compiler::impUnsupportedHWIntrinsic(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig, bool mustExpand)
+{
+    // When we aren't supported we need to return nullptr when attempting to expand inline
+    // (gtIsRecursiveCall == false) as the gtNewMustThrowException node can mess with the
+    // inliner in unexpected ways.
+
+    if (mustExpand)
+    {
+        for (unsigned i = 0; i < sig->numArgs; i++)
+        {
+            impPopStack();
+        }
+
+        return gtNewMustThrowException(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, JITtype2varType(sig->retType),
```

Comment:
Perhaps, parametrizing the exception would be better. We will need other exceptions.

Reply:
What other exceptions are we going to need (I'll parameterize it, just interested as I wasn't aware of others)?

Reply:
Maybe I will need the other exceptions for out of range arguments and non-numeric type arguments, but depends on our future implementation.

Reply:
Ok.

I had thought we were just going to throw PNSE for those as well, but have a more explicit failure makes sense as well.

Reply:
Fixed.